libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -624,39 +624,49 @@ - (void)setOauthClientId:(NSString *)newClientId
-    [self addSidCookieForInstance];
+    __weak typeof(self) weakSelf = self;
+    [self removeCookiesFromDomains:@[@".salesforce.com", @".force.com", @".cloudforce.com"] withCompletion:^{
+        [weakSelf addSidCookieForInstance];
+    }];
 }
 
-+ (void)removeCookies:(NSArray *)cookieNames fromDomains:(NSArray *)domainNames
-{
-    NSAssert(cookieNames != nil && [cookieNames count] > 0, @"No cookie names given to delete.");
++ (void)removeCookiesFromDomains:(NSArray *)domainNames withCompletion:(nullable void(^)())completionBlock {
     NSAssert(domainNames != nil && [domainNames count] > 0, @"No domain names given for deleting cookies.");
-    NSHTTPCookieStorage *cookieStorage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
-    NSArray *fullCookieList = [NSArray arrayWithArray:[cookieStorage cookies]];
-    for (NSHTTPCookie *cookie in fullCookieList) {
-        for (NSString *cookieToRemoveName in cookieNames) {
-            if ([[cookie.name lowercaseString] isEqualToString:[cookieToRemoveName lowercaseString]]) {
-                for (NSString *domainToRemoveName in domainNames) {
-                    if ([[cookie.domain lowercaseString] hasSuffix:[domainToRemoveName lowercaseString]]) {
-                        [cookieStorage deleteCookie:cookie];
-                    }
-                }
-            }
-        }
-    }
+    WKWebsiteDataStore *dateStore = [WKWebsiteDataStore defaultDataStore];
+    NSSet *websiteDataTypes = [NSSet setWithArray:@[ WKWebsiteDataTypeCookies]];
+    [dateStore fetchDataRecordsOfTypes:websiteDataTypes
+                     completionHandler:^(NSArray<WKWebsiteDataRecord *> *records) {
+                         
+                         NSMutableArray<WKWebsiteDataRecord *> *deletedRecords = [NSMutableArray new];
+                         for ( WKWebsiteDataRecord * record in records) {
+                             for(NSString *domainName in domainNames) {
+                                 if ([record.displayName containsString:domainName]) {
+                                     [deletedRecords addObject:record];
+                                 }
+                             }
+                         }
+                         if (deletedRecords.count > 0)
+                             [[WKWebsiteDataStore defaultDataStore] removeDataOfTypes:websiteDataTypes
+                                                                       forDataRecords:deletedRecords
+                                                                    completionHandler:^{
+                                                                        if (completionBlock)
+                                                                            completionBlock();
+                                                                    }];
+                     }];
+    
 }
 
-+ (void)removeAllCookies
++ (void)removeAllCookiesWithCompletion:(void(^)())completionBlock
```

Comment:
Same for this method.  Either keep the original synchronous interface and deprecate it, or do away with it altogether.  Nothing uses this method, and we need to get out of the business of exposing these, one way or another.